docs/tutorials/custom/custom-base-dog.md

```diff
@@ -345,42 +345,58 @@ Name your module `my-custom-base`.
 Enter the path (for example, `/home/fido/robotdog/run.sh`) to your module's executable file in the **Executable path** field.
 Click **Save Config** at the bottom of the page.
 
-![Screenshot of the Viam app CONFIG tab with the Modules subtab open, showing my-custom-base configured.](/tutorials/custom-base-dog/module-config.png)
+![Screenshot of the Viam app Config tab with the Modules subtab open, showing my-custom-base configured.](/tutorials/custom-base-dog/module-config.png)
 
 ## Configure the components
 
 Now that the custom base code is set up, you need to configure all your hardware components.
-Navigate to the **Components** subtab of the **Config** tab.
+Navigate to the **Components** subtab of your robot's **Config** tab.
 
-### Configure the base
+### Configure the camera
 
-In the **Create Component** field, give your base a name.
-We called ours "quadruped".
-In the **Type** drop-down select `base`.
-In the **Model** field, type in `viamlabs:base:robotdog`.
-Click **Create Component**.
+Configure the ribbon camera on the dog as a `webcam` following our [webcam documentation](/components/camera/webcam/).
 
-![Screenshot of the Viam app CONFIG tab with the Config subtab open, showing quadruped configured.](/tutorials/custom-base-dog/config-modular-component.png)
+Click **Save config**.
 
-You need to add the `ip_address` and `port` attributes to your base config.
-In the attributes field, paste the following, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`):
+### Configure the base
+
+Because your custom base relies on a local module, you need to use raw JSON to configure your modular resource.
+Use the **Mode** selector in the upper-left corner of the **Config** tab to switch to **Raw JSON** mode.
+
+Locate the `"components": []` portion of the config file.
+At the end of your camera configuration, add a comma and then add the following base configuration:
 
 ```json
-{
-  "ip_address": "<HOSTNAME>.local",
-  "port": 5001
-}
+    {
+      "namespace": "rdk",
+      "name": "quadruped",
+      "type": "base",
+      "model": "viamlabs:base:robotdog",
+      "attributes": {
+          "ip_address": "<HOSTNAME>.local",
+        "port": 5001
+      },
+      "depends_on": []
+    }
 ```
 
+Edit the `ip_address` attribute to match your robot's IP address, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`).

```

Comment:
Hmm, we have to be careful here, `robotdog.local`, or anything using the word `HOSTNAME`, is _not_ an IP address. I think we have to do a terrible thing:

```suggestion
Edit the `ip_address` attribute to match your robot's hostname, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`).
```

Looking [upstream](https://github.com/viam-labs/robot-dog-module/blob/main/robotdog/src/robotdog.py), yeah it looks like `ip_address` it's just a string with no validation == just a hostname.

Best we can do downstream is use the word "hostname" everywhere we can to describe it.

Reply:
ooh strange ok